### PR TITLE
feat(losses): implement trajectory imitation loss

### PR DESCRIPTION
**Pull request by riita10069**
## Summary

Implements #15 — The primary task loss for AutoE2E training.

## Changes

- **Model/model_components/losses/__init__.py** — losses package
- **Model/model_components/losses/trajectory_loss.py** — `TrajectoryImitationLoss` (nn.Module)
- **Model/tests/test_trajectory_loss.py** — Unit tests

## Design

- Supports `smooth_l1` (default) and `mse` loss types
- Optional temporal weighting: exponential decay `λ^t` where `λ` is configurable (default 1.0 = uniform)
- Input: `trajectory_pred (B, 128)`, `trajectory_target (B, 128)` — 64 timesteps × 2 signals [acceleration, curvature]
- Output: scalar loss

```python
from model_components.losses import TrajectoryImitationLoss

criterion = TrajectoryImitationLoss(loss_type="smooth_l1", temporal_decay=0.98)
loss = criterion(trajectory_pred, trajectory_target)
loss.backward()
```

## Test plan

- [x] Unit tests pass (`test_trajectory_loss.py`)
- [x] Output is scalar
- [x] Gradient flows to input
- [x] Temporal weighting changes loss value
- [x] smooth_l1 vs mse produce different values

## Status: Draft

Tests on `main` are broken due to unrelated flexible backbone changes (#36). Will rebase after #36 is merged.

### CODE REVIEW

**Review by m-zain-khawaja — APPROVED** [review]
Approved - thanks @riita10069
